sharding-orchestration/sharding-orchestration-center/src/main/java/org/apache/shardingsphere/orchestration/center/instance/ApolloInstance.java

```diff
@@ -0,0 +1,209 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.shardingsphere.orchestration.center.instance;
+
+import com.ctrip.framework.apollo.Config;
+import com.ctrip.framework.apollo.ConfigChangeListener;
+import com.ctrip.framework.apollo.ConfigService;
+import com.ctrip.framework.apollo.core.ConfigConsts;
+import com.ctrip.framework.apollo.enums.PropertyChangeType;
+import com.ctrip.framework.apollo.model.ConfigChange;
+import com.ctrip.framework.apollo.model.ConfigChangeEvent;
+import com.ctrip.framework.apollo.openapi.client.ApolloOpenApiClient;
+import com.ctrip.framework.apollo.openapi.client.constant.ApolloOpenApiConstants;
+import com.ctrip.framework.apollo.openapi.dto.NamespaceReleaseDTO;
+import com.ctrip.framework.apollo.openapi.dto.OpenItemDTO;
+import com.google.common.collect.Sets;
+import com.google.common.primitives.Ints;
+import lombok.Getter;
+import lombok.Setter;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.shardingsphere.orchestration.center.api.ConfigCenter;
+import org.apache.shardingsphere.orchestration.center.configuration.InstanceConfiguration;
+import org.apache.shardingsphere.orchestration.center.instance.node.ConfigTreeNode;
+import org.apache.shardingsphere.orchestration.center.listener.DataChangedEvent;
+import org.apache.shardingsphere.orchestration.center.listener.DataChangedEventListener;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Properties;
+import java.util.Set;
+
+/**
+ * Registry center for Apollo.
+ *
+ * @author dongzonglei
+ */
+@Slf4j
+public final class ApolloInstance implements ConfigCenter {
+    
+    private static final String SHARDING_SPHERE_KEY_ROOT = "/";
+    
+    private static final String SHARDING_SPHERE_KEY_SEPARATOR = "/";
+    
+    private static final String APOLLO_KEY_SEPARATOR = ".";
+    
+    private static final String APOLLO_KEY_APP_ID = "app.id";
+    
+    private static final String APOLLO_KEY_ENV = "env";
+    
+    private static final String APOLLO_KEY_CLUSTER = ConfigConsts.APOLLO_CLUSTER_KEY;
+    
+    private static final String APOLLO_KEY_META = ConfigConsts.APOLLO_META_KEY;
+    
+    private String namespace;
+    
+    private String appId;
+    
+    private String env;
+    
+    private String clusterName;
+    
+    private String administrator;
+    
+    private Config apolloConfig;
+    
+    private ApolloOpenApiClient client;
+    
+    private ConfigTreeNode tree = new ConfigTreeNode(null, "/", Sets.<ConfigTreeNode>newHashSet());
+    
+    @Getter
+    @Setter
+    private Properties properties = new Properties();
+    
+    @Override
+    public void init(final InstanceConfiguration config) {
+        initApolloConfig(config);
+        initApolloOpenApiClient();
+        initKeysRelationship();
+    }
+    
+    private void initApolloConfig(final InstanceConfiguration config) {
```

Comment:
Shall we rename the method name, as the method doesn't only initialize the "apollyConfig".

Reply:
This is only a `private` method and inner used. The `public` method is `init(final InstanceConfiguration config)`.